### PR TITLE
Governor: enableCountingStrategy

### PR DESCRIPTION
**Pull request by mrmemes-eth**
This sets a mapping of counting strategy selector to a boolean representing whether it's a valid choice as a proposal counting strategy.

### CODE REVIEW

File: src/governor/GovernorCoreFacet.sol [hunk 1]
```diff
@@ -96,6 +96,15 @@ contract GovernorCoreFacet is AccessControl, IEIP712, IGovernor {
         return GovernorStorage.isProposalTokenEnabled(proposalToken);
     }
 
+    /**
+     * @notice determine if a counting strategy is valid to configure a proposal for voting with.
+     * @param countingStrategy the address of the counting strategy to check
+     * @return true if the counting strategy is a proposal token, false otherwise
+     */
+    function isCountingStrategyEnabled(bytes4 countingStrategy) public view returns (bool) {
```
Comment (desi) [inline src/governor/GovernorCoreFacet.sol:104]: The doc says the "address" of the counting strategy but the param is a bytes4 so not an address. Can you help me understand?

Comment (mrmemes-eth) [reply src/governor/GovernorCoreFacet.sol:104]: doc's wrong :D

Comment (mrmemes-eth) [reply src/governor/GovernorCoreFacet.sol:104]: fixed

File: src/utils/GovernorDiamondInit.sol [hunk 2]
```diff
@@ -73,6 +73,8 @@ contract GovernorDiamondInit {
         config.proposalThresholdToken = proposalThresholdToken;
         // by default, we only allow the default proposal token to be used for proposals
         config.proposalTokens[proposalToken] = true;
+        // by default, we only allow the simple counting strategy to be used for proposals
+        config.countingStrategies[0x6c4b0e9f] = true;
```
Comment (desi) [inline src/utils/GovernorDiamondInit.sol:77]: Where does this `0x6c4b0e9f` come from?

Comment (mrmemes-eth) [reply src/utils/GovernorDiamondInit.sol:77]: > Where does this `0x6c4b0e9f` come from?

```solidity
bytes4(keccak256("simpleWeight(uint256)"))
```

Comment (mrmemes-eth) [reply src/utils/GovernorDiamondInit.sol:77]: It's also used a bit further up in the file and could be swapped out for either that `bytes4` incantation or `GovernorCoreFacet.simpleWeight.selector`.

**Review by desi — APPROVED** [review]
I had a few questions but otherwise it looks good to me!
🚀

File: src/utils/GovernorDiamondInit.sol (hunk 2 above)
Comment (desi) [reply src/utils/GovernorDiamondInit.sol:77]: Thank you for the explaination!

**Pull request merged**